config/config.go

```diff
@@ -80,6 +82,28 @@ func LoadGenesisConfig(filename string) (*GenesisConfig, error) {
 	return &nc, nil
 }
 
+func (gc *GenesisConfig) ComputeGenesisHash() (types.Hash, error) {
+	hasher := ktypes.NewHasher()
+
+	hasher.Write([]byte(gc.ChainID))
+	hasher.Write(gc.Leader)
+
+	// sort the validators by public key
+	sort.Sort(ktypes.ByPubKey(gc.Validators))
+	for _, v := range gc.Validators {
+		hasher.Write(v.PubKey)
+		binary.Write(hasher, binary.BigEndian, v.Power)
+	}
+
+	binary.Write(hasher, binary.BigEndian, gc.MaxBlockSize)
+	binary.Write(hasher, binary.BigEndian, gc.JoinExpiry)
+	binary.Write(hasher, binary.BigEndian, gc.VoteExpiry)
+	binary.Write(hasher, binary.BigEndian, gc.DisabledGasCosts)
+	binary.Write(hasher, binary.BigEndian, gc.MaxVotesPerTx)
```

Comment:
One point about these, but not necessarily needing change: these are almost too convenient in that we can casually change the type of these and this code will not break, but the result will change, which is a consensus breaker.  It's a lot more cumbersome to use the functions like binary.BigEndian.Append/PutUint64 etc, so let's go ahead with this now since we will certainly be changing `gc` a lot before release, but at some point I think we should harden this code.